src/main/java/dataStructures/hashSet/openAddressing/HashSet.java

```diff
@@ -185,71 +216,33 @@ public int capacity() {
      * within the valid range of the array bounds.
      * This ensures that the index falls within the range of available buckets.
      *
-     * @param element the element to be hashed
+     * @param element    the element to be hashed
+     * @param collisions the number of collisions so far.
      * @return the bucket index where the element should be placed
      */
-    private int hashFunction(T element) {
-        int hashCode = element.hashCode();
-        return (hashCode & 0x7FFFFFFF) % buckets.length;
-    }
+    private int hashFunction(T element, int collisions) {
+        int hashCode = element.hashCode() & 0x7FFFFFFF;
 
-    /**
-     * Given an element, returns the index of an empty (defined as null OR tombstone) bucket to insert the element at.
-     * If the element is already present in the HashSet, return its index.
-     *
-     * @param element the given element to probe an empty bucket for.
-     * @return the index of an empty bucket.
-     */
-    private int linearProbe(T element) {
-        int startingProbeIndex = hashFunction(element);
-
-        int currentBucketIndex = startingProbeIndex;
-        for (int i = 0; i < this.capacity() - 1; i++) {
-            T existingElement = this.buckets[currentBucketIndex];
-            // check for empty / available bucket.
-            if (this.isEmptyBucket(currentBucketIndex)) {
-                return currentBucketIndex;
-            }
+        // This step is where the OA and chaining implementation differs.
+        int probeAdjustedHash = linearProbe(hashCode, collisions);
 
-            // check if element is equals to the element in the bucket.
-            // Checks equality of element using Object::equals and Object::hashCode
-            if (element.equals(existingElement)
-                && element.hashCode() == existingElement.hashCode()) {
-                return currentBucketIndex;
-            }
-            currentBucketIndex = (currentBucketIndex + 1) % this.capacity();
-        }
-        return ELEMENT_NOT_FOUND; // placeholder return value for now. Will never reach this line.
+        return probeAdjustedHash % capacity(); // Division-method. Not the most ideal.
```

Comment:
i was debating whether to leave this line here to throw it to be the job of the probe function. It does feel like almost every probe function should have this %, and so put it here instead. But im kinda leaning towards still being the job of the WhateverProbe function / class. Reason being, i can define a function that probes without ever going it out of range.
While having the additional % here wouldnt affect correctness (since not out of range), it does feel slightly weird.

lmk if u think otherwise

Reply:
I can see where you are coming from! In the notes:
<img width="429" alt="image" src="https://github.com/4ndrelim/data-structures-and-algorithms/assets/79049665/083d1c75-fe48-48ce-8404-a8a961acebd8">

specifically `h(k, i) = h(k, 1) + i mod m`, it seems that the probe function will have the `%` as seen by `i mod m`. I will see if its more feasible/clearer to abstract that out.

Reply:
ohh this slide is to be interpreted as (h(k, 1) + i) mod m otherwise u might exceed the capacity m

Reply:
Oh, if that is the intended interpretation, then should I still proceed with moving the modulo into the `linearProbe` function? Because the current implementation will be interpreted as intended already.

Reply:
hmm ok yeah leave it as it is